Sources/SecureXPC/XPCMachClient.swift

```diff
@@ -63,114 +63,29 @@ import Foundation
-    public func sendMessage<M: Encodable, R: Decodable>(_ message: M,
-                                                        route: XPCRouteWithMessageWithReply<M, R>,
-                                                        withReply reply: @escaping XPCReplyHandler<R>) throws {
-        let encoded = try Request(route: route.route, payload: message).dictionary
-        sendWithReply(encoded: encoded, withReply: reply)
-    }
-    
-    /// Does the actual work of sending an XPC message which receives a reply.
-    private func sendWithReply<R: Decodable>(encoded: xpc_object_t,
-                                             withReply reply: @escaping XPCReplyHandler<R>) {
-        xpc_connection_send_message_with_reply(createConnection(), encoded, nil, { xpcResponse in
-            let result: Result<R, XPCError>
-            if xpc_get_type(xpcResponse) == XPC_TYPE_DICTIONARY {
-                do {
-                    let response = try Response(dictionary: xpcResponse)
-                    if response.containsPayload {
-                        result = Result.success(try response.decodePayload(asType: R.self))
-                    } else if response.containsError {
-                        result = Result.failure(try response.decodeError())
-                    } else {
-                        result = Result.failure(.unknown)
-                    }
-                } catch let error as XPCError  {
-                    result = Result.failure(error)
-                } catch {
-                    result = Result.failure(.unknown)
-                }
-            } else if xpc_equal(xpcResponse, XPC_ERROR_CONNECTION_INVALID) {
-                result = Result.failure(.connectionInvalid)
-            } else if xpc_equal(xpcResponse, XPC_ERROR_CONNECTION_INTERRUPTED) {
-                result = Result.failure(.connectionInterrupted)
-            } else if xpc_equal(xpcResponse, XPC_ERROR_TERMINATION_IMMINENT) {
-                result = Result.failure(.terminationImminent)
-            } else { // Unexpected
-                result = Result.failure(.unknown)
-            }
-            reply(result)
-        })
-    }
-    
-    /// Creates and returns connection for the mach service stored by this instance of the client.
-    private func createConnection() -> xpc_connection_t {
-        let connection = xpc_connection_create_mach_service(self.machServiceName, nil, 0)
-        xpc_connection_set_event_handler(connection, { (event: xpc_object_t) in
-            // A block *must* be set as the handler, even though this block does nothing.
-            // If it were not set, a crash would occur upon calling xpc_connection_resume.
-        })
-        xpc_connection_resume(connection)
-        
-        return connection
-    }
+public class XPCMachClient: XPCBaseClient {
+
+	// This client implementation intentionally does not store a reference to the xpc_connection_t as it can become
```

Comment:
I'll move it to the `createConnection()` method, where I think it'll be closer to where it'll be expected.

I think keeping it replicated between the two kinds of client is kinda worth it, in case we change this behaviour on a case-by-case basis. 

WDYT?